### ISSUE TITLE
[heroku]: Method of adding invoices to a bookkeeping system

### ISSUE THREAD

**Issue by thijsc** [heroku/roadmap#104]
### Required Terms

- [X] I agree to follow this project's [Code of Conduct](https://github.com/heroku/roadmap/blob/main/CODE_OF_CONDUCT.md)
- [X] I have read and accept the [Salesforce Program Agreement](https://www.salesforce.com/company/program-agreement/)

### What service(s) is this request for?

heroku

### Tell us about what you're trying to solve. What challenges are you facing?

```markdown
A problem we are facing is that I have to personally log into Heroku every quarter to download a few invoices.

All European companies are required to add those to their books. The invoices are not sent via email, so I can't automate this. I also can't give our bookkeeper access since it requires full admin permission to open an invoice. We need a way to either receive the invoices on an email address, or have a person that does not have full admin permissions to access them.
```

**Comment by kouroubel** [comment]
This is similar to #53 which I proposed sometime ago...

**Comment by thijsc** [comment]
> This is similar to https://github.com/heroku/roadmap/issues/53 which I proposed sometime ago...

I saw that one, I think the use case is slightly different. This mostly about only the admin having the permissions to download invoices. Maybe one fix could handle both use cases though.

**Comment by kouroubel** [comment]
The context is slightly different, true, but the essence is the same: we need a team role that can access financial information without having admin privileges.

**Comment by thijsc** [comment]
I'd be fine with centralizing this in one issue. Could somebody from Heroku chime in?

**Comment by afawcett** [comment]
@thijsc and @kouroubel thanks both for your collaboration and sharing your needs. Firstly, yes we can merge these two items, I will get that done. A theme developing in our roadmap asks is more granular access controls. I think grouping these togethers and reviewing them holistically will help - so I'll add a tag here as well. 

As workaround, though it would require a small Heroku app, the Invoice download facility is available via the Heroku API, so its possible build a small web app with access only to your admins that uses this API https://devcenter.heroku.com/articles/platform-api-reference#invoice.

**Comment by thijsc** [comment]
I'd be ok with running that if you can provide an app like that.

**Comment by afawcett** [comment]
@thijsc just to confirm your saying if we shared perhaps an open source solution you would be happy deploying it to your own Heorku account? Such an approach would not come with support - more like a use as-is and of course if open source contribute back etc. Is this what you had in mind? If so, its an interesting approach to delivering some features that I am interested in, not a good fit for all of course. Thoughts?

**Comment by thijsc** [comment]
Interesting idea, def a possible approach to deliver more niche features. However I do think that this case is not niche at all, literally every single EU based customer you have needs a way to get to the invoices.

**Comment by afawcett** [comment]
@thijsc I do agree - just offering an interim option - thanks again for sharing this need.

**Comment by afawcett** [comment]
@ArturT yes, from the UI, you can login to Heroku Dashboard and go to your user profile, click the Billing tab to explore past invoices. It will present you with a HTML page for each month and you can turn that into a PDF from your browser.

**Comment by ArturT** [comment]
@afawcett I was hoping I could automate that by downloading Invoice PDF from your API. Probably you meant Heroku Dashboard instead of Heroku API when you said:

> the Invoice download facility is available via the Heroku API

**Comment by ArturT** [comment]
@patryk-smc I agree.

**Comment by ArturT** [comment]
@afawcett Could you prioritize this? It would be awesome if we could define a billing email address in Heroku billing settings and then the PDF invoice could be sent to it. Thank you.

An alternative solution (less useful) would be to have a billing role https://github.com/heroku/roadmap/issues/53 so a user can access only billing (to download invoices) without access to apps.
